packages/material-ui/src/Tabs/Tabs.js

```diff
@@ -2,96 +2,217 @@ import * as React from 'react';
 import { isFragment } from 'react-is';
 import PropTypes from 'prop-types';
 import clsx from 'clsx';
-import { refType } from '@material-ui/utils';
+import { refType, deepmerge } from '@material-ui/utils';
+import { unstable_composeClasses as composeClasses } from '@material-ui/unstyled';
+import experimentalStyled from '../styles/experimentalStyled';
+import useThemeProps from '../styles/useThemeProps';
+import useTheme from '../styles/useTheme';
 import debounce from '../utils/debounce';
 import ownerWindow from '../utils/ownerWindow';
 import { getNormalizedScrollLeft, detectScrollType } from '../utils/scrollLeft';
 import animate from '../internal/animate';
 import ScrollbarSize from './ScrollbarSize';
-import withStyles from '../styles/withStyles';
-import TabIndicator from './TabIndicator';
 import TabScrollButton from '../TabScrollButton';
 import useEventCallback from '../utils/useEventCallback';
-import useTheme from '../styles/useTheme';
+import tabsClasses, { getTabsUtilityClass } from './tabsClasses';
 
-export const styles = (theme) => ({
-  /* Styles applied to the root element. */
-  root: {
-    overflow: 'hidden',
-    minHeight: 48,
-    WebkitOverflowScrolling: 'touch', // Add iOS momentum scrolling.
-    display: 'flex',
-  },
-  /* Styles applied to the root element if `orientation="vertical"`. */
-  vertical: {
-    flexDirection: 'column',
-  },
-  /* Styles applied to the flex container element. */
-  flexContainer: {
-    display: 'flex',
+const useUtilityClasses = (styleProps) => {
+  const {
+    vertical,
+    fixed,
+    hideScrollbar,
+    scrollableX,
+    scrollableY,
+    flexContainerVertical,
+    centered,
+    scrollButtonsHideMobile,
+    classes,
+  } = styleProps;
+
+  const slots = {
+    root: ['root', vertical && 'vertical'],
+    scroller: [
+      'scroller',
+      fixed && 'fixed',
+      hideScrollbar && 'hideScrollbar',
+      scrollableX && 'scrollableX',
+      scrollableY && 'scrollableY',
+    ],
+    flexContainer: [
+      'flexContainer',
+      flexContainerVertical && 'flexContainerVertical',
+      centered && 'centered',
+    ],
+    indicator: ['indicator'],
+    scrollButtons: ['scrollButtons', scrollButtonsHideMobile && 'scrollButtonsHideMobile'],
+  };
+
+  return composeClasses(slots, getTabsUtilityClass, classes);
+};
+
+const TabsRoot = experimentalStyled(
+  'div',
+  {},
+  {
+    name: 'MuiTabs',
+    slot: 'Root',
+    overridesResolver: (props, styles) => {
+      const { styleProps } = props;
+
+      return deepmerge(
+        {
+          ...(styleProps.vertical && styles.vertical),
+          ...styles.scrollButtons,
+          ...(styleProps.scrollButtonsHideMobile && styles.scrollButtonsHideMobile),
+        },
+        styles.root || {},
+      );
```

Comment:
```suggestion
      return {
        ...styles.root,
        ...styles.scrollButtons,
        ...(styleProps.vertical && styles.vertical),
        ...(styleProps.scrollButtonsHideMobile && styles.scrollButtonsHideMobile),
      };
```

No need for deepmerge anymore. I hope prettier won't complain 😄

Reply:
Also you will probably need to remove the import for `deepmerge` forgot about that one :)